circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -276,14 +276,21 @@ def _append_measurement_circuit(
     if not inplace:
         qc = qc.copy()
 
+    # If the circuit has no measurements, the Sampler will fail.  So, we
+    # measure one qubit as a temporary workaround to
+    # https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/issues/422
+    pauli_indices = cog.pauli_indices
+    if not pauli_indices:
```

Comment:
What can `pauli_indices` be here? Should we just explicitly state what we expect? (e.g. `if pauli_indices == []`)

Or we could just assert `pauli_indices == []` inside the if?

Reply:
> What can pauli_indices be here? Should we just explicitly state what we expect? (e.g. if pauli_indices == [])

It's a sequence of ints.  It contains the qubit indices that need to be measured.

The choice of style here is influenced by [PEP 8](https://peps.python.org/pep-0008/):

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false:
> ```python
> # Correct:
> if not seq:
> if seq:
> ```
> ```python
> # Wrong:
> if len(seq):
> if not len(seq):
> ```
The "nice" thing about doing it this way is it doesn't matter what type the sequence is.  For instance, if it were to be a tuple of indices, then it will still behave correctly with the current code, but it would not if we were to compare with `[]` because `() != []`.

Reply:
nice, thanks for explanation

Reply:
Wow, this is more profound than I thought. Love it!